pkg/api/store/clustertemplate/store.go

```diff
@@ -157,3 +177,39 @@ func isDefaultTemplateRevision(apiContext *types.APIContext, id string) (bool, e
 
 	return false, nil
 }
+
+func (p *Store) canSetEnforce(apiContext *types.APIContext, data map[string]interface{}, templateID string) error {
+	//check if turning on enforced flag
+
+	enforcedFlagInData := convert.ToBool(data[managementv3.ClusterTemplateFieldEnforced])
+	enforcedFlagChanged := enforcedFlagInData
+
+	if templateID != "" {
+		var template managementv3.ClusterTemplate
+		if err := access.ByID(apiContext, apiContext.Version, managementv3.ClusterTemplateType, templateID, &template); err != nil {
+			return err
+		}
+		if template.Enforced != enforcedFlagInData {
+			enforcedFlagChanged = true
+		}
+	}
+	if enforcedFlagChanged {
+		//only admin can set the flag
+		ma := gaccess.MemberAccess{
+			Users:     p.Users,
+			GrLister:  p.GrLister,
+			GrbLister: p.GrbLister,
+		}
+		//if user is admin, no checks needed
+		callerID := apiContext.Request.Header.Get(gaccess.ImpersonateUserHeader)
+		isAdmin, err := ma.IsAdmin(callerID)
+		if err != nil {
+			return err
+		}
+		if !isAdmin {
+			return httperror.NewAPIError(httperror.PermissionDenied, fmt.Sprintf("ClusterTemplate's %v field cannot be changed", managementv3.ClusterTemplateFieldEnforced))
```

Comment:
Can we change the error message to show that non-admin user made this request hence enforcement can't be set?

Reply:
Yeah, deliberately keeping the error message generic, reason is to not reveal any unnecessary rbac information.